Sources/XCTest/Public/XCTAssert.swift

```diff
@@ -171,10 +171,10 @@ public func XCTAssertEqual<T: Equatable>(_ expression1: @autoclosure () throws -
     }
 }
 
-public func XCTAssertEqual<T: FloatingPoint>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> T, accuracy: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+public func XCTAssertEqual<T: Strideable & Numeric>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> T, accuracy: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
```

Comment:
I've done some local experimentation with the version of this code which is included in Xcode, and I believe it's possible to simplify this code to _only_ require conformance to the `Numeric` protocol and remove the `Strideable` requirement. I think the latter is needed _currently_ because the code uses `distance(to:)` to calculate differences, but I'm not sure if that way of getting the difference is strictly necessary. If we distill the core "are two numbers equal w/accuracy" calculation down to a helper function, I think that can look something like this:

```swift
private func areEqual<T : Numeric>(_ exp1: T, _ exp2: T, accuracy: T) -> Bool {
    if exp1 == exp2 {
        return true
    } else {
        // NaN values are handled implicitly, since the <= operator returns false when comparing any value to NaN.
        let difference = (exp1.magnitude > exp2.magnitude) ? exp1 - exp2 : exp2 - exp1
        return difference.magnitude <= accuracy.magnitude
    }
}
```

And this removes the need for `distance(to:)` and anything else which currently requires `Strideable`.

Could we employ this technique in this PR too, so we can change the type requirements to only need `Numeric`? (Or, if anyone knows a strong reason to _keep_ the current implementation, let me know.) Thanks!